src/Neo/Ledger/Blockchain.cs

```diff
@@ -339,15 +339,27 @@ private VerifyResult OnNewExtensiblePayload(ExtensiblePayload payload)
             return VerifyResult.Succeed;
         }
 
-        private VerifyResult OnNewTransaction(Transaction transaction)
+        private VerifyResult OnNewTransaction(Transaction tx)
         {
-            switch (system.ContainsTransaction(transaction.Hash))
+            UInt256 hash;
+            try
+            {
+                // Avoid serialization problems
+                hash = tx.Hash;
```

Comment:
Just a question, could you please clarify, are these serialization problems connected with transaction's Signers or with something else?

Reply:
Any fields if it's null

Reply:
Than I presume that the same issue is relevant not only for transactions but also for blocks (`OnNewBlock`, we also access the block's hash in this method) and may be for Inventory (but need to check, from the first glance it seems that everything is OK with this payload).

Reply:
Yes, it could be done also with blocks

Reply:
Then we probably need to fix it for blocks as far.

Reply:
Done